sdk/core/core/test/cmocka/test_span.c

```diff
@@ -250,97 +250,110 @@ static void az_span_find_beginning_success()
-  // token: "defg" (span+3)
-  span = out_span;
-
-  token = az_span_token(span, delim, &out_span);
-  assert_true(az_span_ptr(token) == az_span_ptr(span));
-  assert_true(az_span_length(token) == 4);
-  assert_true(az_span_capacity(token) == az_span_length(token));
-  assert_true(az_span_ptr(out_span) == (az_span_ptr(span) + az_span_length(token) + az_span_length(delim)));
-  assert_true(az_span_length(out_span) == (az_span_length(span) - az_span_length(token) - az_span_length(delim)));
-  assert_true(az_span_capacity(out_span) == (az_span_capacity(span) - az_span_capacity(token) - az_span_capacity(delim)));
-
-  // token: "defg" (span+10)
-  span = out_span;
-
-  token = az_span_token(span, delim, &out_span);
-  assert_true(az_span_ptr(token) == az_span_ptr(span));
-  assert_true(az_span_length(token) == 4);
-  assert_true(az_span_capacity(token) == az_span_length(token));
-  assert_true(az_span_ptr(out_span) == (az_span_ptr(span) + az_span_length(token) + az_span_length(delim)));
-  assert_true(az_span_length(out_span) == (az_span_length(span) - az_span_length(token) - az_span_length(delim)));
-  assert_true(az_span_capacity(out_span) == (az_span_capacity(span) - az_span_capacity(token) - az_span_capacity(delim)));
-
-  // token: "defg" (span+17)
-  span = out_span;
-
-  token = az_span_token(span, delim, &out_span);
-  assert_true(az_span_ptr(token) == az_span_ptr(span));
-  assert_true(az_span_length(token) == 4);
-  assert_true(az_span_capacity(token) == az_span_length(token));
-  assert_true(az_span_ptr(out_span) == NULL);
-  assert_true(az_span_length(out_span) == 0);
-  assert_true(az_span_capacity(out_span) == 0);
-
-  // Out_span is empty.
-  span = out_span;
-  
-  token = az_span_token(span, delim, &out_span);
-  assert_true(az_span_is_content_equal(token, AZ_SPAN_NULL));
+static void az_span_find_target_more_chars_than_prefix_of_source_fails()
+{
+  az_span span = AZ_SPAN_FROM_STR("abcd");
+  az_span target = AZ_SPAN_FROM_STR("zab");
+
+  assert_int_equal(az_span_find(span, target), -1);
+}
+
+static void az_span_find_overlapping_target_success()
+{
+  az_span span = AZ_SPAN_FROM_STR("abcdefghij");
+  az_span target = az_span_slice(span, 6, 9);
+
+  assert_int_equal(az_span_find(span, target), 6);
+}
+
+static void az_span_find_embedded_NULLs_success()
+{
+  az_span span = AZ_SPAN_FROM_STR("abcd\0\0fghij");
+  az_span target = AZ_SPAN_FROM_STR("\0\0");
+
+  assert_int_equal(az_span_find(span, target), 4);
+}
+
+static void az_span_find_capacity_checks_success()
+{
+  uint8_t* buffer = (uint8_t*)"aaaa";
+
+  assert_int_equal(az_span_find(az_span_init(buffer, 2, 4), az_span_init(buffer, 2, 3)), 0);
+  assert_int_equal(az_span_find(az_span_init(buffer, 2, 3), az_span_init(buffer, 2, 4)), 0);
+  assert_int_equal(az_span_find(az_span_init(buffer, 2, 3), az_span_init(buffer, 0, 2)), 0);
+  assert_int_equal(az_span_find(az_span_init(buffer, 0, 2), az_span_init(buffer, 0, 2)), 0);
+}
+
+static void az_span_find_overlapping_checks_success()
```

Comment:
nit: Should we merge this test with `az_span_find_overlapping_target_success`. They are very similar in intent.

Also, given we are now also testing the "failure" condition of returning -1, maybe rename the test with the merge and remove `success` from it.